cloud/pkg/dynamiccontroller/application/application.go

```diff
@@ -239,12 +241,22 @@ func (a *Application) Close() {
 		return
 	}
 
+	a.tim = time.Now()
 	a.count--
 	if a.count == 0 {
 		a.Status = Completed
 	}
 }
 
+func (a *Application) LastCloseTime() time.Time {
+	a.countLock.Lock()
+	defer a.countLock.Unlock()
+	if a.count == 0 && !a.tim.Equal(time.Time{}) {
```

Comment:
better a.tim.IsZero()

Reply:
and plz check a.tim != nil

Reply:
what dose "check a.tim != nil" mean?  the default value of a.tim is time.Time{}, a.tim.IsZero() can detect empty situations?

Reply:
1. For code robustness, better check a.tim !=nil.
2. yes

Reply:
I have tried it and will report an error `Cannot convert 'nil' to type 'Time' `

Reply:
my error, it is struct but not point.